autoscale_cloudroast/test_repo/autoscale/system/integration/test_system_integration_rcv3.py

```diff
@@ -650,6 +665,116 @@ def test_scale_up_down_on_rcv3_and_clb(self):
+                                                 server_list_after_up[0])
+
+        post_del_servers, post_del_nodes = self._verify_rcv3_group_state(
+            pool_group,
+            pool_group.launchConfiguration.loadBalancers[0].loadBalancerId,
+            as_server_count,
+            expected_rcv3_server_count - 1,
+            as_server_count)
+
+        # Construct, create, and execute a policy to scale back to min.
+        policy_down_data = {'change': -scale_amt, 'cooldown': 0}
+        expected_rcv3_server_count = server_count_before_autoscaling
+        as_server_count = as_server_count - scale_amt
+
+        self.asb.create_policy_webhook(pool_group.id,
+                                       policy_down_data,
+                                       execute_policy=True)
+
+        self._verify_rcv3_group_state(
+            pool_group,
+            pool_group.launchConfiguration.loadBalancers[0].loadBalancerId,
+            as_server_count,
+            expected_rcv3_server_count,
+            as_server_count)
+
+        # Confirm that after scaling back to zero, the lb state matches the
+        # inital state
+        final_node_list = [n.id for n in self.rcv3_client.get_nodes_on_pool(
+            self.pool.id).entity.nodes]
+        self.assertEquals(set(initial_node_list), set(final_node_list),
+                          msg='The set of nodes on the pool is incorrect'
+                          'after scaling')
+
+    def _verify_rcv3_group_state(self, group, pool_id,
+                                 num_as, num_rc, num_nova):
+        """
+        Verify that the expected state of rcv3, autoscale, and nova is reached
+        """
+        server_list = []
+        # Since zero servers is an exit condtion that raises and exception in
+        # wait_for_expected_number_of_servers, if the expected number is zero
+        # just check the group state.
+        if num_as == 0:
+            self.wait_for_expected_group_state(group.id, num_as)
+        else:
+            server_list = self.asb.\
+                wait_for_expected_number_of_active_servers(group.id,
+                                                           num_as,
+                                                           timeout=600,
+                                                           asserter=self)
+        # Wait for RackConnect to reflect Otter's preferred configuration.
+        node_list = self.asb.wait_for_expected_number_of_active_servers(
+            pool_id, num_rc, timeout=300, api="RackConnect",
+            asserter=self)
+        # When checking for zero nova servers, we use the deletion specific
+        # function in order to provide a longer wait time for servers to
+        # delete completely.
+        if num_nova == 0:
+            self.assert_servers_deleted_successfully(
+                group.launchConfiguration.server.name)
+        else:
+            self.verify_server_count_using_server_metadata(group.id, num_nova)
```

Comment:
Same goes here too. `verify_server_count_using_server_metadata` _can_ expect 0 which leads to the question if `assert_servers_deleted_successfully` can be implemented by using `verify_server_count_using_server_metadata` but that can be done later.